docs/failsafe-controller.md

```diff
@@ -0,0 +1,192 @@
+| Never           | Does not invoke the customized action but instead moves on to the next recorded action to respond to the attempt | It becomes 0 |
+| Forever         | Repeats the customized action for every subsequent attempts of a given execution | Will repeat forever |
+
+Actions are recorded by starting a sequence using one of the static methods in the `net.jodah.failsafe.Actions` class and chaining them together. Each of the resulting sequence can
+then be recorded for a particular execution. For example:
+```
+  Actions.waitTo("connect").before().returning(true);
+```
+In this example, the first Failsafe attempt for the execution would be blocked until some other part of the code decides to notify the "connect" condition and return true as a result of that attempt.
+```
+  Actions.doThrow(NullPointerException)
+    .then().waitTo("connect again").before().returning(false)
+    .then().doReturn(true);
+```
+In this example, the first Failsafe attempt for the next execution would result in a java.lang.NullPointerException being thrown back. The expectation is that a second attempt would be made at which point it would block until some other part of the code decides to notify the "connect again" at which point it would return false. When a third attempt is made by Failsafe, true would then be returned and no more attempts would be expected for this execution.
+
+The above example can be written in Spock as:
+```
+ doThrow(NullPointerException)
+    .then().waitTo("connect again").before().returning(false)
+    .then().doReturn(true)
+```
+or with more Groovy flair as:
+```
+ doThrow(NullPointerException) + waitTo("connect again").before().returning(false) + doReturn(true)
+```
+### Execution
+As indicated above, an execution encompass the point where Failsafe is called using one of the methods where a task is provided to Failsafe. The execution is completed when the `java.util.concurrent.Future` for asynchronous tasks is completed and can provide the result for the task or the error that occurred or again when Failsafe return with a result or an exception for synchronous tasks.  Whenever an execution starts, the Failsafe controller will retrieve the sequence of actions associated with the next recorded execution. An execution is recorded with the controller using one of the `net.jodah.failsafe.FailsafeController.onNextExecution(net.jodah.failsafe.Actions.Done)` method. For Groovy developers, it is also possible to use a closure that returns the sequence of actions using the `net.jodah.failsafe.FailsafeController.onNextExecution(groovy.lang.Closure)` method. No matter which way you do it, the sequence of actions will be defined as indicated in the previous section.
```

Comment:
s/b "...an execution encompasses..."
s/b "...one of the `net.jodah.failsafe.FailsafeController.onNextExecution(net.jodah.failsafe.Actions.Done)` methods."